lambda/copy_cuboid_lambda.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2016 The Johns Hopkins University Applied Physics Laboratory
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+event: {
+    's3_key': (str) S3 object key of source cuboid.
+    'translate': (list[x, y, z]) Translate cuboid's position by these deltas.
+                                 Must be a multiple of the cuboid's dimensions.
+    'region': (str) AWS region resources exist in.
+    'lookup_key': (str) coll_id&exp_id&chan_id (DB ids with separating ampersands).
+    'ingest_job': (int) Id of ingest job created for this copy.
+    'version': (int) 0 - should always be 0 until versioning actually implementd.
+    'object_store_config': (dict) Initialization data for AWSObjectStore.
+    {
+        'id_count_table': '...',
+        'page_in_lambda_function': '...',
+        'id_index_table': '...',
+        'cuboid_bucket': '...',
+        's3_index_table': '...',
+        'page_out_lambda_function': '...',
+        's3_flush_queue': '...'
+    }
+}
+"""
+
+import boto3
+import hashlib
+import logging
+from spdb.c_lib.ndtype import CUBOIDSIZE
+from spdb.c_lib.ndlib import XYZMorton, MortonXYZ
+from spdb.spatialdb.object import AWSObjectStore
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+def handler(event, context):
+    """
+    Entry point for the lambda function.
+
+    Args:
+        event (dict): Lambda input parameters.  See module level comments.
+        context (dict): Standard lambda context parameter (ignored).
+    """
+    copy_cuboid(event)
+
+def copy_cuboid(event):
+    """
+    Main worker function for this lambda.
+
+    Args:
+        event (dict): Lambda input parameters.  See module level comments.
+    """
+    validate_translation(event['translate'])
```

Comment:
Why do we need to validate the dimensions when we are copying from one channel to another?  Also what if we get an error?  Does the valueError that is thrown have enough information for us to know what key caused the issue?

Reply:
Yes, the translation should be the same for all cuboids, so it will fail for all. This is just an additional sanity check. I actually have the same check done in the script that invokes this lambda, but I'm trying to code defensively.